lib/GenTest/Validator/ResultsetComparator.pm

```diff
@@ -43,13 +43,21 @@ sub validate {
 	return STATUS_WONT_HANDLE if $results->[0]->query() =~ m{ANALYZE}sio;
 
 	if ( ($compare_outcome == STATUS_LENGTH_MISMATCH) ||
-	     ($compare_outcome == STATUS_CONTENT_MISMATCH) 
+	     ($compare_outcome == STATUS_CONTENT_MISMATCH)
 	) {
 		say("---------- RESULT COMPARISON ISSUE START ----------");
 	}
 
 	if ($compare_outcome == STATUS_LENGTH_MISMATCH) {
-		if ($query =~ m{^\s*select}io) {
+		# before the select, we can have:
+		# 1. a hint /*+ ... */
+		# 2. horizontal whitespace (excludes newlines)
+		# 3. EXPLAIN (...)
+		# 4. EXPLAIN ONE_WORD (e.g. EXPLAIN ANALYZE)
+		# These can repeat any number of times.
+		#                  ____1_____  2   ________3________   ______4_______
+		#                 |          | |  |                 | |              |
+		if ($query =~ m{^(\/\*\+.*\*\/|\h|EXPLAIN\h*(\(.*\))?|EXPLAIN\h+[a-z]+)*select}io) {
```

Comment:
Would be a bit simpler if you strip off the hints and EXPLAIN separately then test if it starts with SELECT. e.g.: (This one just ignores the hints.) https://github.com/yugabyte/randgen/blob/cb25f4435f634b0b803a329df8941f2db78e29be/lib/GenTest/Mixer.pm#L152

Reply:
consider modifying the regexp to use non-greedy matching.
```
my $is_select = ($query =~ s{/\*.+?\*/}{}sgor) =~ m{^\s*SELECT}sio;
```

Above handles the case when there are multiple comments.